src/utils/mcrypto.ts

```diff
@@ -0,0 +1,9 @@
+import bcrypt from 'bcryptjs';
+
+export const comparePwd = async (pwd: string, hash: string): Promise<boolean> => {
+  return bcrypt.compare(pwd, hash);
+};
+
+export const hashPwd = async (pwd: string, salt = 10): Promise<string> => {
+  return bcrypt.hash(pwd, salt);
```

Comment:
The `salt` param here is not a real salt. Also, real salt should be persisted to database.

https://github.com/dcodeIO/bcrypt.js/blob/master/src/bcrypt.js#L164

Reply:
@bsdelf 
OK, This `salt` is really not a real SALT, Success `salt length` may be more suitable.
However, the actual SALT is generated according to this value, and the comments of the method of the method are mentioned `Salt length to generate or salt to use`, 
https://github.com/dcodeIO/bcrypt.js/blob/master/src/bcrypt.js#L156

Before executing `_hash`, call `bcrypt.genSalt` to generate the corresponding SALT value;
https://github.com/dcodeIO/bcrypt.js/blob/master/src/bcrypt.js#L168

In addition, the password encrypted data exists in the database. Based on this data, you can get the corresponding SALT through the `bcrypt.genSalt` method, so there is no unsusted problem.
https://github.com/dcodeIO/bcrypt.js/blob/master/src/bcrypt.js#L297

Reply:
eg:
Created a test case of HashPWD, the code is as follows:
```
import bcrypt from 'bcryptjs';
describe('test hashPwd', () => {
  const pwd = '123456';
  const saltLen = 10;
  test('salt to save', async () => {
    const hashPwd = await bcrypt.hash(pwd, saltLen);
    const saltHashPwd = hashPwd.slice(0, 29);
    console.log('save to database in password, saltHashPwd:', saltHashPwd);
    const salt = await bcrypt.getSalt(hashPwd);
    console.log('bcrypt get salt, salt:', salt);
    expect(salt).toBe(saltHashPwd);
  });
});
```
The running results are as follows:
![image](https://user-images.githubusercontent.com/12078330/167542465-0fbfe5d8-d896-47c9-8323-fea75d519845.png)